src/profilers/profile-serializer.ts

```diff
@@ -64,12 +64,16 @@ export function serializeTimeProfile(
    * @param stack - the stack trace to the current node.
    */
   function serializeNode(root: TimeProfileNode) {
-    let nodes = [root];
-    let stacks: number[][] = [[]];
-    while (nodes.length > 0) {
-      let node = nodes.pop();
-      let stack = stacks.pop();
-      if (node !== undefined && stack !== undefined) {
+    let entries: {node: TimeProfileNode, stack: number[]}[] = [];
```

Comment:
nit: declaring this as an interface rather than an inline type definition would be better.